File: 2021/src/2.js
const { getInputLines } = require('./util/input')

const input = getInputLines().map(l => l.split(' '))

const solve = (input, p2 = false) => {
  let horizontal = 0
  let depth = 0
  let aim = 0

  for (const step of input) {
    const direction = step[0]
    const length = parseInt(step[1])

    switch (direction) {
      case 'forward':
        horizontal += length
        if (p2) {
          depth += aim * length
        }
        break
      case 'up':
        if (p2) {
          aim -= length
        } else {
          depth -= length
        }
        break
      case 'down':
        if (p2) {
          aim += length
        } else {
          depth += length
        }
        break
      default:
        break
    }
  }

  return horizontal * depth
}

console.log('P1:', solve(input))
console.log('P2:', solve(input, true))

File: 2015/src/2.py
from util.input import get_input_lines


def get_wrapping_paper(dims: str) -> int:
    x, y, z = sorted([int(x) for x in dims.split("x")])
    return (2 * x * y) + (2 * x * z) + (2 * y * z) + (x * y)


def get_ribbon(dims: str) -> int:
    x, y, z = sorted([int(x) for x in dims.split("x")])
    return (x * y * z) + (2 * x + 2 * y)


data = get_input_lines()
p1 = sum([get_wrapping_paper(dims) for dims in data])
p2 = sum([get_ribbon(dims) for dims in data])

print("P1:", p1)
print("P2:", p2)
